crates/forge_main/src/ui.rs

```diff
@@ -97,14 +98,18 @@ impl<F: API> UI<F> {
         Event::new(EVENT_USER_HELP_QUERY, content)
     }
 
-    pub fn init(cli: Cli, api: Arc<F>) -> Result<Self> {
+    pub async fn init(cli: Cli, api: Arc<F>) -> Result<Self> {
         // Parse CLI arguments first to get flags
-
         let env = api.environment();
+        // load the workflow.
+        let workflow = api.load(cli.workflow.as_deref()).await?;
+        let forge_command_manager = ForgeCommandManager::from(workflow.commands);
```

Comment:
Load the commands everytime. This way new commands won't require a restart of the shell. We already load the workflow everytime the user inputs anything.